### PR TITLE
No more nasty errors during flutter upgrade/channel on Windows

### LINKED ISSUE #14578: Error when flutter upgrade or flutter channel modifies flutter.bat
On my Windows RS2 (v10.0.14393) machine, previously running the alpha channel, I attempted to switch to the beta channel. Error message as below...

```
[C:\Users\timsneath] flutter doctor
[√] Flutter (on Microsoft Windows [Version 10.0.14393], locale en-US, channel alpha)
    • Flutter version 0.0.21 at c:\git\flutter
    • Framework revision 2e449f06f0 (10 days ago), 2018-01-29 14:26:51 -0800
    • Engine revision 6921873c71
    • Tools Dart version 2.0.0-dev.16.0
    • Engine Dart version 2.0.0-edge.da1f52592ef73fe3afa485385cb995b9aec0181a

[√] Android toolchain - develop for Android devices (Android SDK 27.0.2)
    • Android SDK at C:\Users\timsneath\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Android Studio (version 3.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Connected devices
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)


[C:\Users\timsneath] flutter channel
Flutter channels:
* alpha
  beta
  dev
  master

[C:\Users\timsneath] flutter channel beta
Switching to flutter channel 'beta'...
git: Branch 'beta' set up to track remote branch 'beta' from 'origin'.
git: Switched to a new branch 'beta'
'ot_pathflutter_tools_dirscript_path"' is not recognized as an internal or external command,
operable program or batch file.
Downloading package sky_engine...                     0.5s
Downloading common tools...                           0.4s
Downloading windows-x64 tools...                      1.0s
Downloading windows-x64 tools...                      1.3s
Downloading android-arm-profile/windows-x64 tools... 3650.2s
Downloading android-arm-release/windows-x64 tools...  0.4s
Downloading android-x86 tools...                      0.7s
Downloading android-x64 tools...                      0.8s
Downloading android-arm tools...                      0.6s
Downloading android-arm-profile tools...              0.5s
Downloading android-arm-release tools...              0.5s
Switching to flutter channel 'beta'...
git: M  VERSION
git: Your branch is up to date with 'origin/beta'.
git: Already on 'beta'

[C:\Users\timsneath]
```

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/14578.

### CODE REVIEW

**Comment by goderbauer** [comment]
I believe the logic around exit code 253 (outdated snapshot of Flutter Tools) is no longer necessary because we control which dart version ships with Flutter and whenever we change that version we create a new snapshot. Therefore, the snapshot should never be outdated. The logic is a relict from a time when Flutter used whatever Dart version was installed on the system.

File: bin/flutter.bat [hunk 1]
```diff
@@ -166,20 +166,15 @@ GOTO :after_subroutine
...
-  )
-  CALL "%dart%" %FLUTTER_TOOL_ARGS% "%snapshot_path%" %*
-  SET exit_code=%ERRORLEVEL%
-)
+REM Chaining the call to 'dart' and 'exit' with an ampersand ensures that
+REM Windows reads both commands into memory once before executing them. This
+REM avoids nasty errors that may otherwise occure when the dart command (e.g. as
+REM part of 'flutter upgrade') modifies this batch script while it is executing.
+REM
+REM Do not use the CALL command in the next line to execute Dart. CALL causes
```
Comment (aam) [inline bin/flutter.bat]: Whoa, who would have thought! Is this documented anywhere?

Comment (goderbauer) [reply bin/flutter.bat]: I did not find any official documentation around this. There was one stack overflow answer explaining this behavior, but I can't find it now... :(

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**